### ISSUE TITLE
Visual Studio crashes detected for StackOverflowException in FlexClassifier.GetClassificationSpans

### ISSUE THREAD

**Issue by olegtk** [GraviKK/FlexHighlighter#2]
I'm an engineer on the Visual Studio team and I got a bug about Visual Studio crashes caused by StackOverflowException in the editor's classifier. From the Watson dump I see it's caused by FlexClassifier.GetClassificationSpans().
Unfortunately I don't have much info to provide as I have no access to symbols. There are only 21 crashes detected, also it's possible that they happened while you were developing FlexHighlighter, so just FYI.

**Comment by ttarasiewicz** [comment]
Thank you for your feedback. Can you tell us in what situation this error occurs?

**Comment by olegtk** [comment]
Not really, the only dump we have is not actionable. Also StackOverflowException is hard to analyze post factum. All I know is user edited .l file, the VS editor calls FlexClassifier.GetClassificationSpans() and StackOverflowException occurs. I'll let you know if I get any additional data.
